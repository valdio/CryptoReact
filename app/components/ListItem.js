import GLOBALS from '../lib/globals'
import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {
    StyleSheet,
    Text,
    View,
} from 'react-native'
import {Link} from 'react-router-native'

export default class ListItem extends Component {
    render() {
        const {fiatCurrency, currencyItem} = this.props
        return (
            <Link
                underlayColor={GLOBALS.COLOR.HEADER_COLOR}
                to={`/details/${currencyItem.id}`} style={styles.container}>
                <View>
                    <View style={styles.section}>
                        <Text style={styles.text}>Symbol: {currencyItem.symbol}</Text>
                        <Text style={styles.text}>Rank: {currencyItem.rank}</Text>
                    </View>
                    <View style={styles.section}>
                        <Text
                            style={styles.text}>Price: {fiatCurrency === GLOBALS.FIAT_CURRENCY.EUR ? `€ ${currencyItem.price_eur}`
                            : fiatCurrency === GLOBALS.FIAT_CURRENCY.CNY ? `¥ ${currencyItem.price_cny}`
                                : `$ ${currencyItem.price_usd}`
                        }</Text>
                        <Text style={styles.text}>Change (24h): {currencyItem.percent_change_24h}%</Text>
                    </View>
                </View>
            </Link>
        )
    }
}

ListItem.propTypes = {
    fiatCurrency: PropTypes.oneOf([GLOBALS.FIAT_CURRENCY.USD, GLOBALS.FIAT_CURRENCY.EUR, GLOBALS.FIAT_CURRENCY.CNY]),
    currencyItem: PropTypes.object.isRequired
}

const styles = StyleSheet.create({
    container: {
        justifyContent: 'center',
        flexDirection: 'column',
        backgroundColor: GLOBALS.COLOR.RED,
        margin: 8,
        padding: 6,
        elevation: 6,
        borderRadius: 6,
        shadowColor: GLOBALS.COLOR.BLACK,
        shadowOffset: {width: 0, height: 4},
        shadowOpacity: 0.8,
        shadowRadius: 2,
    },
    section: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        margin: 8,
    },
    text: {
        flex: 0.5,
        fontSize: 16,
        color: GLOBALS.COLOR.WHITE
    }
})
