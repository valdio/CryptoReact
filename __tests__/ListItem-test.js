import React from 'react'
import ListItem from '../app/components/ListItem'

import renderer from 'react-test-renderer'

const usdCryptocurrency = {
    'id': 'bitcoin',
    'name': 'Bitcoin',
    'symbol': 'BTC',
    'rank': '1',
    'price_usd': '10838.8',
    'price_btc': '1.0',
    '24h_volume_usd': '9145340000.0',
    'market_cap_usd': '182874531426',
    'available_supply': '16872212.0',
    'total_supply': '16872212.0',
    'max_supply': '21000000.0',
    'percent_change_1h': '0.35',
    'percent_change_24h': '0.71',
    'percent_change_7d': '30.99',
    'last_updated': '1518964166'
}

/**
 * Snapshot Test with JEST
 */

test('<ListItem/> Snapshot Test', () => {
    const tree = renderer.create(
        <ListItem
            fiatCurrency='USD'
            currencyItem={usdCryptocurrency}
        />).toJSON()
    expect(tree).toMatchSnapshot()
})




