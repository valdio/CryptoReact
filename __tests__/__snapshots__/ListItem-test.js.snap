// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem/> Snapshot Test 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FF473A",
      "borderRadius": 6,
      "elevation": 6,
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 8,
      "padding": 6,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={null}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 0.5,
            "fontSize": 16,
          }
        }
      >
        Symbol: 
        BTC
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 0.5,
            "fontSize": 16,
          }
        }
      >
        Rank: 
        1
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 0.5,
            "fontSize": 16,
          }
        }
      >
        Price: 
        $ 10838.8
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 0.5,
            "fontSize": 16,
          }
        }
      >
        Change (24h): 
        0.71
        %
      </Text>
    </View>
  </View>
</View>
`;
